saleor/warehouse/tests/test_stock_management.py

```diff
@@ -574,3 +593,229 @@ def test_deallocate_stock_for_order(
     allocations = order_line.allocations.all()
     assert allocations[0].quantity_allocated == 0
     assert allocations[1].quantity_allocated == 0
+
+
+@mock.patch("saleor.plugins.manager.PluginsManager.product_variant_back_in_stock")
+def test_increase_stock_with_back_in_stock_webhook_not_triggered(
+    product_variant_back_in_stock_webhook, allocation
+):
+    stock = allocation.stock
+    stock.quantity = 10
+    stock.save(update_fields=["quantity"])
+
+    increase_stock(allocation.order_line, stock.warehouse, 50, allocate=False)
+
+    stock.refresh_from_db()
+    assert stock.quantity == 60
+
+    flush_post_commit_hooks()
+    product_variant_back_in_stock_webhook.assert_not_called()
+
+
+@mock.patch("saleor.plugins.manager.PluginsManager.product_variant_out_of_stock")
+def test_decrease_stock_with_out_of_stock_webhook_triggered(
+    product_variant_out_of_stock_webhook_mock, allocation
+):
+    stock = allocation.stock
+    stock.quantity = 50
+    stock.save(update_fields=["quantity"])
+    allocation.quantity_allocated = 50
+    allocation.save(update_fields=["quantity_allocated"])
+    warehouse_pk = allocation.stock.warehouse.pk
+
+    decrease_stock(
+        [
+            OrderLineData(
+                line=allocation.order_line,
+                quantity=50,
+                variant=stock.product_variant,
+                warehouse_pk=warehouse_pk,
+            )
+        ],
+    )
+
+    flush_post_commit_hooks()
+
+    product_variant_out_of_stock_webhook_mock.assert_called_once()
+
+
+@mock.patch("saleor.plugins.manager.PluginsManager.product_variant_out_of_stock")
+def test_decrease_stock_with_out_of_stock_webhook_triggered_among_warehouses(
```

Comment:
this test is way too long

Reply:
We need this test. We have to be sure for 100% that webhooks will recognize changes only for a given channel. But I've decided to conduct some refactor. Please, let me know if this is fine for you.